src/main/java/com/google/roomies/database/FirebaseDatabase.java

```diff
@@ -0,0 +1,158 @@
+package com.google.roomies;
+
+import static com.google.roomies.ProjectConstants.PROJECT_ID;
+import static com.google.roomies.ListingRequestParameterNames.TIMESTAMP;
+
+import com.google.api.core.ApiFuture;
+import com.google.api.core.ApiFutures;
+import com.google.auth.oauth2.GoogleCredentials;
+import com.google.cloud.firestore.CollectionReference;
+import com.google.cloud.firestore.DocumentReference;
+import com.google.cloud.firestore.DocumentSnapshot;
+import com.google.cloud.firestore.FieldValue;
+import com.google.cloud.firestore.Firestore;
+import com.google.cloud.firestore.QueryDocumentSnapshot;
+import com.google.cloud.firestore.QuerySnapshot;
+import com.google.cloud.firestore.WriteResult;
+import com.google.common.annotations.VisibleForTesting;
+import com.google.firebase.cloud.FirestoreClient;
+import com.google.firebase.FirebaseApp;
+import com.google.firebase.FirebaseOptions;
+import java.io.IOException;
+import java.util.List;
+import java.util.Map; 
+
+public class FirebaseDatabase implements Database {
```

Comment:
You might consider moving this into a database package, and making the constructor package-private. In this factory-interface world, non of the database clients should be trying to instantiate a FirebaseDatabase directly, and in fact you would like to be able to prevent them from doing so (on purpose or accidentally).